tests/vmi_lifecycle_test.go

```diff
@@ -1186,17 +1186,74 @@ var _ = Describe("[rfe_id:273][crit:high][arm64][vendor:cnv-qe@redhat.com][level
 			})
 
 			It("[test_id:3203]the vmi with cpu.features that cannot match nfd labels on a node should not be scheduled", func() {
+				var featureDenyList = map[string]struct{}{
+					"svm": {},
+				}
+				appendFeatureFromFeatureLabel := func(supportedFeatures []string, label string) []string {
+					if strings.Contains(label, services.NFD_CPU_FEATURE_PREFIX) {
+						feature := strings.TrimPrefix(label, services.NFD_CPU_FEATURE_PREFIX)
+						if _, exist := featureDenyList[feature]; !exist {
+							return append(supportedFeatures, feature)
+						}
+					}
+					return supportedFeatures
+				}
 
+				removeDups := func(elements []string) (intersection []string) {
+					found := make(map[string]struct{})
+					for _, element := range elements {
+						if _, exist := found[element]; !exist {
+							intersection = append(intersection, element)
+							found[element] = struct{}{}
+						}
+					}
+					return intersection
+				}
+
+				setIntersection := func(firstSet, secondSet []string) []string {
+					firstSetMap := make(map[string]struct{})
+					var setOfFeaturesWithDups []string
+
+					for _, element := range firstSet {
+						firstSetMap[element] = struct{}{}
+					}
+
+					for _, element := range secondSet {
+						if _, exist := firstSetMap[element]; exist {
+							setOfFeaturesWithDups = append(setOfFeaturesWithDups, element)
+						}
+					}
+					return removeDups(setOfFeaturesWithDups)
+				}
+
+				GetSupportedCPUFeaturesFromNodes := func(nodes k8sv1.NodeList) []string {
+					var supportedFeatures []string
+					for label := range nodes.Items[0].Labels {
```

Comment:
you'll need to assure `nodes.Items` is not empty, right ? (I'm aware it's probably stupid to run the test is your node list is empty ... :) )

IIUC, here you're just initializing the list of supported features. Would it make sense to generalize this into a single for ?

Reply:
Hey ,
I will check that the nodes list isn't empty np although the tests will fail much earlier in that case ,i think that they won't run at all.

In that function i perform sets intersection of many sets and if i'll initial the first set to be empty the result will be empty 
because intersection with empty set is empty set to i initialed it to the first set of labels.

I can do it with boolean but i think that it will be much more complicated and less understandable

Reply:
Honestly i can't think on A clearer alternative but if anyone have idea i'll be happy to hear.
Also don't forget this is an anonymous function  that run only in one test and we usually 
run the tests with not more than 10 nodes with less than 200 labels so the performance 
shouldn't be a problem in that case.